src/viewProviders/resources.ts

```diff
@@ -671,9 +666,22 @@ export async function loadLocalResources(): Promise<
 }
 
 export async function loadDirectResources(): Promise<DirectEnvironment[]> {
-  // fetch all direct connections and their resources; each connection will be treated the same as a
-  // CCloud environment (connection ID and environment ID are the same)
-  const directEnvs = await getDirectResources();
+  // load all direct connection IDs and specs from storage (as the source of truth)
+  const directConnectionMap: DirectConnectionsById =
+    await getResourceManager().getDirectConnections();
+
+  // fetch all direct connections and their resources by connection ID, where each connection will
+  // be treated the same as a CCloud environment (connection ID and environment ID are the same)
+  const graphqlPromises: Promise<DirectEnvironment | undefined>[] = Array.from(
+    directConnectionMap.keys(),
+  ).map((id) => getDirectResources(id));
+  // filter out any undefined results (which shouldn't happen, but just in case)
+  const directEnvs: DirectEnvironment[] = await Promise.all(graphqlPromises).then(
+    (results): DirectEnvironment[] =>
+      results.filter((env): env is DirectEnvironment => env !== undefined),
+  );
```

Comment:
This area was the only real use case for using the [`directConnections` GraphQL query](https://github.com/confluentinc/vscode/blob/9bd1dee6100c893c5a966742ce5cdf521528f1ef/src/graphql/sidecar.graphql#L115-L116). I erred on the side of moving away from it entirely and requiring every GraphQL query provide a connection ID since the sidecar should never know about a connection that the extension isn't already tracking in secret storage.